newIDE/app/src/EffectsList/index.js

```diff
@@ -426,7 +426,25 @@ export default function EffectsList(props: Props) {
   );
 
   const all3DEffectMetadata = React.useMemo(
-    () => allEffectMetadata.filter(effect => effect.isMarkedAsOnlyWorkingFor3D),
+    () => {
+      const effectMetadata = allEffectMetadata.filter(
+        effect => effect.isMarkedAsOnlyWorkingFor3D
+      );
+      const lightEffectMetadata = effectMetadata.filter(effect =>
+        effect.type.endsWith('Light')
+      );
+      const fogEffectMetadata = effectMetadata.filter(effect =>
+        effect.type.endsWith('Fog')
+      );
+      const otherEffectMetadata = effectMetadata.filter(
```

Comment:
is it worth changing this with a single list browsing with a reduce?

Reply:
It was a bit too verbose with a reduce, but I used a for loop.